src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -277,6 +285,46 @@ private void InitializeFilterList(UserSettings settings)
             }
         }
 
+        private void UpdateUpgradeProjectVisibility()
+        {
+            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+            {
+                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
do you need to switch to main thread here? can you just move this switch to the place where it is absolutely necessary?

Reply:
it absolutely needs to switch to UI thread here. can discuss in person if you have questions